sunpy/net/dataretriever/tests/test_norh.py

```diff
@@ -1,41 +1,135 @@
-     ),
-    (TimeRange('2012/12/1', '2012/12/2'),
-     'ftp://anonymous:data@sunpy.org@solar-pub.nao.ac.jp/pub/nsro/norh/data/tcx/2012/12/tca121201',
-     'ftp://anonymous:data@sunpy.org@solar-pub.nao.ac.jp/pub/nsro/norh/data/tcx/2012/12/tca121202'
-     ),
-    (TimeRange('2012/3/7', '2012/3/14'),
-     'ftp://anonymous:data@sunpy.org@solar-pub.nao.ac.jp/pub/nsro/norh/data/tcx/2012/03/tca120307',
-     'ftp://anonymous:data@sunpy.org@solar-pub.nao.ac.jp/pub/nsro/norh/data/tcx/2012/03/tca120314'
-     )
-])
-def test_get_url_for_time_range(timerange, url_start, url_end):
-    urls = norh.NoRHClient()._get_url_for_timerange(timerange, wavelength=17*u.GHz)
+def test_fetch_working():
+    """
+    Tests if the mock fetch contains the correct data.
+    """
+
+    qr1 = NORHClient.search(a.Time('2012/10/4', '2012/10/6'),
+                            a.Instrument('norh'), a.Wavelength(17*u.GHz))
+
+    # Create a mock query object
+    mock_qr = mock_query_object('2012/10/4', '2012/10/6', wavelength=17*u.GHz)
+
+    mock_qr = mock_qr[0]
+    qr = qr1[0]
+    # Assert the values
+    assert mock_qr.source == qr.source
+    assert mock_qr.instrument == qr.instrument
+    assert mock_qr.physobs == qr.physobs
+    assert mock_qr.provider == qr.provider
+    assert mock_qr.url == qr.url
+    assert mock_qr.time == qr.time
+
+    # Assert if the time range is same
+    assert qr1.time_range() == TimeRange('2012/10/4', '2012/10/7')
+
+    # Assert the fetch object, and whether it returns the correct set of files
+    res = NORHClient.fetch(qr1)
+    download_list = res.wait(progress=False)
+    assert len(download_list) == len(qr1)
+
+    download_list.sort()
+    # Assert each of the files have the same file names
+    assert download_list[0].split('/')[-1] == 'tca121004'
+    assert download_list[1].split('/')[-1] == 'tca121005'
+    assert download_list[2].split('/')[-1] == 'tca121006'
+
+
+@mock.patch('sunpy.net.dataretriever.sources.norh.NoRHClient._get_url_for_timerange',
+            return_value=create_url('2012/3/7', '2012/3/14', wavelength=17*u.GHz))
+def test_get_url_for_time_range(mock_get_url):
+    urls = norh.NoRHClient()._get_url_for_timerange(TimeRange(parse_time('2012/3/7'),
+                                                              parse_time('2012/3/14')),
+                                                    wavelength=17*u.GHz)
+    assert isinstance(urls, list)
+    assert urls[0] == 'ftp://anonymous:data@sunpy.org@solar-pub.nao.ac.jp/pub/nsro/norh/data/tcx/2012/03/tca120307'
+    assert urls[-1] == 'ftp://anonymous:data@sunpy.org@solar-pub.nao.ac.jp/pub/nsro/norh/data/tcx/2012/03/tca120314'
+
+# Wavelength 34 GHz
+@mock.patch('sunpy.net.dataretriever.sources.norh.NoRHClient._get_url_for_timerange',
+            return_value=create_url('2012/3/7', '2012/3/14', wavelength=34*u.GHz))
+def test_get_url_for_time_range_34(mock_get_url):
```

Comment:
same in here, we should have them parametrized. You [can see an example](https://github.com/hackebrot/pytest-tricks/issues/32#issuecomment-461380532) on how it would look.

Reply:
Done